doc/manual/manual.tex

```diff
@@ -1595,14 +1741,198 @@ \subsubsection{Compressible models}
+
+\item
+``implicit reference density profile'':
+\[
+ \nabla \cdot \textbf{u} + \frac{1}{\bar{\rho}} \frac{\partial \bar{\rho}}{\partial z} \frac{\textbf{g}}{g} \cdot \textbf{u} = 0,
+\]
+which uses the same approximation for the density as ``reference density profile'', 
+but implements this term on the left-hand side instead of the right-hand side of the mass
+conservation equation. This effectively uses the current velocity $\textbf{u}$ instead
+of an extrapolated velocity from the last timesteps.
+
+ \item 
+``ask material model'', which uses ``isothermal compression'' if the material model is compressible 
+(as reported by the \texttt{is\_compressible()} function) and ``incompressible'' otherwise.
+\end{itemize}
+
+
+\paragraph*{On the stress tensor approximation.}
+
+Incompressibility in the mass conservation equation automatically simplifies the shear strain rate in the momentum and temperature equation from
+\[
+ \nabla \cdot \tau = 
+ \nabla \cdot \left[
+ 2\eta \left(\varepsilon(\mathbf u)
+                                  - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1\right)
+                                  \right]
+\]
+to
+\[
+  \nabla \cdot \tau =  \nabla \cdot \left[
+ 2\eta \varepsilon(\mathbf u) \right].
+\]
+
+\subsubsection{Temperature equation approximation}
+
+The density occurs multiple times in the temperature equation. Depending on the selected approximation it is computed in two different ways:
+
+\begin{itemize}
+ \item 
+``real density'': Use the full density as computed for the buoyancy term of the momentum equation (i.e. the return value of the material model for ``density``),
+
+\item
+``reference density profile'': Use the density as computed for the reference profile (which can be constant, an adiabatic profile, or an entirely different function, and is determined by the adiabatic conditions model).
+\end{itemize}
+
+\subsubsection{Approximation of the buoyancy term}
+The buoyancy term (right-hand side of the momentum equation) always uses the 
+density that is provided by the material model (see Section~\ref{sec:material-models}). 
+There, one can decide to use a density that for example depends on temperature 
+and pressure, or one that depends on temperature and depth; and one can also 
+decide to use density deviations from a reference state instead of a full density 
+(see Section~\ref{sec:pressure-static-dyn}).
+
+\subsubsection{Reference state: The adiabatic profile}
+
+The reference temperature profile $\bar{T}$, reference density profile $\bar{\rho}$
+and the reference pressure $\bar{p}$ are computed in the adiabatic conditions model 
+(provided by the class \textit{AdiabaticConditionds}, see Section~\ref{sec:adiabatic}). 
+By default, these fields satisfy adiabatic conditions (if adiabatic heating is included 
+in the model):
+\begin{align}
+  \rho C_p \frac{\textrm{d} \bar{T}(z)}{\textrm{d}z} 
+  &=
+  \frac{\partial\bar\rho}{\partial \bar{T}} \bar{T}(z) g_z,
```

Comment:
should be `\frac{\partial\bar\rho}{\partial T}` (not `\bar T` in the denominator).